User changing its own password

tonijurjevic96 opened the pull request:
Allow users to change their own password

Usage: SET PASSWORD TO newPassword REPLACE oldPassword;

tonijurjevic96 commented:
@kgolubic

kgolubic commented:
As RN I will use:

- Added functionality that allows each user to change their own password. 
[#2097](https://github.com/memgraph/memgraph/pull/2097)

Josipmrden commented on tests/e2e/user_changing_its_own_password/common.py:
```diff
@@ -0,0 +1,54 @@
+    cursor = connection.cursor()
+    execute_and_fetch_all(cursor, "USE DATABASE memgraph")
+    try:
+        execute_and_fetch_all(cursor, "DROP DATABASE clean")
```
why do we need this database "clean"?

Josipmrden commented on tests/e2e/user_changing_its_own_password/user_changing_its_own_password.py:
```diff
@@ -0,0 +1,58 @@
+    execute_query("CREATE USER IF NOT EXISTS testuser IDENTIFIED BY 'testpassword';", AUTH)
+    AUTH = ("testuser", "testpassword")
+    execute_query("SET PASSWORD TO 'newPassword' REPLACE 'testpassword';", AUTH)
+    print("Created user.")
```
remove print, you can type just assert True

Josipmrden commented on tests/e2e/user_changing_its_own_password/user_changing_its_own_password.py:
```diff
@@ -0,0 +1,58 @@
+    try:
+        AUTH = ("testuser", "newPassword")
+        execute_query("SHOW DATABASE SETTINGS;", AUTH)
+        # execute_query("MATCH n RETURN n LIMIT 1;")
```
remove this execute_query comments

Josipmrden commented on tests/e2e/user_changing_its_own_password/user_changing_its_own_password.py:
```diff
@@ -0,0 +1,58 @@
+
+        execute_query("SHOW DATABASE SETTINGS;", AUTH)
+
+        # execute_query("SET PASSWORD TO 'newPassword' REPLACE 'testpassword';")
```
remove comments

Josipmrden commented on src/glue/auth_handler.cpp:
```diff
@@ -16,6 +16,7 @@
 #include <fmt/format.h>
 
 #include "auth/auth.hpp"
+#include "auth/crypto.hpp"
```
see if this header is necessary

tonijurjevic96 commented:
Please delete either the [master < EPIC] or [master < Task] part, depending on what are your needs.

[master < Epic] PR
- [x] Write E2E tests
- [x] Compare the [benchmarking results](https://bench-graph.memgraph.com/) between the master branch and the Epic branch
- [x] Provide the full content or a guide for the final git message
    - Added functionality that allows each user to change its own password


### CI Testing Labels
Please select the appropriate CI test labels _(CI -build=**build-name** -test=**test-suite**)_


### Documentation checklist
- [x] Add the documentation label tag
- [x] Add the bug / feature label tag
- [x] Add the milestone for which this feature is intended
    - If not known, set for a later milestone
- [x] Write a release note, including added/changed clauses
    - Added functionality that allows each user to change its own password
      Usage:
      SET PASSWORD TO newPassword REPLACE oldPassword; 
- [x] Link the documentation PR here
    - https://github.com/memgraph/documentation/pull/819
- [x] Tag someone from docs team in the comments
